app/services/cocina_generator/structural/generator.rb

```diff
@@ -27,36 +27,34 @@ def generate
       # 2) Rebuild filesets that have attached_files that are preserved
       # 3) Add new filesets that have attached_files in the local store.
       def build_filesets
-        filesets = find_preserved_filesets.to_h do |filename, fileset|
```

Comment:
I noticed that we were generating a hash in `find_preserved_filesets` and then immediately calling `.to_h` on that again. Now we call `.to_h` on it once down within `find_preserved_filesets`. This change combines two bits of code.

Reply:
If we're just calling `.values`, it is necessary to create a hash?

Reply:
@justinlittman Good eye. That led me to a little more refactoring which I think brings added clarity. 🍻

Reply:
Much appreciated. That area of the code is very difficult to grok.